democracy/views/hearing.py

```diff
@@ -134,32 +133,41 @@ def get_queryset(self):
                 to_attr='main_section_list'
             )
         )
+
+        # unless the user is a superuser, only show open hearings in the hearing list
+        if not self.request.user.is_superuser:
+            queryset = queryset.filter(open_at__lte=now())
+
         return self.common_queryset_filtering(queryset)
 
     def get_object(self):
         id_or_slug = self.kwargs[self.lookup_url_kwarg or self.lookup_field]
 
-        try:
-            queryset = self.common_queryset_filtering(Hearing.objects.with_unpublished())
-            queryset = queryset.prefetch_related(
-                Prefetch(
-                    'sections',
-                    queryset=Section.objects.filter(type__identifier='main'),
-                    to_attr='main_section_list'
-                )
+        queryset = self.common_queryset_filtering(Hearing.objects.with_unpublished())
+        queryset = queryset.prefetch_related(
+            Prefetch(
+                'sections',
+                queryset=Section.objects.filter(type__identifier='main'),
+                to_attr='main_section_list'
             )
+        )
+
+        try:
             obj = queryset.get_by_id_or_slug(id_or_slug)
         except Hearing.DoesNotExist:
             raise NotFound()
 
-        user = self.request.user
-        is_superuser = user and user.is_authenticated() and user.is_superuser
-
+        is_superuser = self.request.user.is_superuser
+        preview_code = None
         if not obj.published and not is_superuser:
             preview_code = self.request.query_params.get('preview')
             if not preview_code or preview_code != obj.preview_code:
                 raise NotFound()
 
+        # require preview_code or superuser status to show a not yet opened hearing
+        if not (preview_code or is_superuser or obj.open_at <= now()):
```

Comment:
162-165 checks if preview code exists and is valid, this line excludes not yet opened hearings in case there is no preview_code or superuser. It might be possible to combine this into 162-165 ifs somehow, but I think this approach is also ok.